scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/Resources.java

```diff
@@ -63,16 +65,15 @@ private Resources() {
                 .build();
     }
 
-    public static List<Integer> selectTwoPortsFromRange(List<Protos.Resource> offeredResources) {
-        List<Integer> ports = new ArrayList<>();
-        offeredResources.stream().filter(resource -> resource.getType().equals(org.apache.mesos.Protos.Value.Type.RANGES))
-                .forEach(resource -> resource.getRanges().getRangeList().stream().filter(range -> ports.size() < 2).forEach(range -> {
-                    ports.add((int) range.getBegin());
-                    if (ports.size() < 2 && range.getBegin() != range.getEnd()) {
-                        ports.add((int) range.getBegin() + 1);
-                    }
-                }));
-        return ports;
+    public static List<Integer> selectTwoUnprivilegedPortsFromRange(List<Protos.Resource> offeredResources) {
+        return offeredResources.stream()
+                .filter(resource -> resource.getType().equals(Protos.Value.Type.RANGES))
+                .flatMap(resource -> resource.getRanges().getRangeList().stream())
+                .flatMapToInt(range -> IntStream.rangeClosed((int) range.getBegin(), (int) range.getEnd()))
+                .filter(port -> port > 1024)
```

Comment:
Ports below 1024 are agreed to be privileged, http://www.w3.org/Daemon/User/Installation/PrivilegedPorts.html
> The TCP/IP port numbers below 1024 are special in that normal users are not allowed to run servers on them. This is a security feaure, in that if you connect to a service on one of these ports you can be fairly sure that you have the real thing, and not a fake which some hacker has put up for you.

Long story short, it means that on n*x systems only root can listen on them. It'll never change, so I thought it didn't make sense to make a static field from it.

Logstash should change the 5000 to be in the privileged port range too.